ovs/lib/setup.py

```diff
@@ -1201,35 +1415,47 @@ def _demote_node(cluster_ip, master_ip, cluster_name, ip_client_map, unique_id,
 
-        external_etcd = EtcdConfiguration.get('/ovs/framework/external_etcd')
-        if external_etcd is None:
-            print 'Leaving Etcd cluster'
-            logger.info('Leaving Etcd cluster')
-            EtcdInstaller.shrink_cluster(master_ip, cluster_ip, 'config', offline_node_ips)
+        try:
+            external_etcd = EtcdConfiguration.get('/ovs/framework/external_etcd')
+            if external_etcd is None:
+                print 'Leaving Etcd cluster'
+                logger.info('Leaving Etcd cluster')
+                EtcdInstaller.shrink_cluster(master_ip, cluster_ip, 'config', offline_node_ips)
+        except Exception as ex:
+            SetupController._print_log_error('leave etcd cluster', ex)
 
         print 'Update configurations'
         logger.info('Update configurations')
-        if unconfigure_memcached is True:
-            endpoints = EtcdConfiguration.get('/ovs/framework/memcache|endpoints')
-            endpoint = '{0}:{1}'.format(cluster_ip, 11211)
-            if endpoint in endpoints:
-                endpoints.remove(endpoint)
-            EtcdConfiguration.set('/ovs/framework/memcache|endpoints', endpoints)
-        if unconfigure_rabbitmq is True:
-            endpoints = EtcdConfiguration.get('/ovs/framework/messagequeue|endpoints')
-            endpoint = '{0}:{1}'.format(cluster_ip, 5672)
-            if endpoint in endpoints:
-                endpoints.remove(endpoint)
-            EtcdConfiguration.set('/ovs/framework/messagequeue|endpoints', endpoints)
+        try:
+            if unconfigure_memcached is True:
+                endpoints = EtcdConfiguration.get('/ovs/framework/memcache|endpoints')
+                endpoint = '{0}:{1}'.format(cluster_ip, 11211)
+                if endpoint in endpoints:
+                    endpoints.remove(endpoint)
+                EtcdConfiguration.set('/ovs/framework/memcache|endpoints', endpoints)
+            if unconfigure_rabbitmq is True:
+                endpoints = EtcdConfiguration.get('/ovs/framework/messagequeue|endpoints')
+                endpoint = '{0}:{1}'.format(cluster_ip, 5672)
+                if endpoint in endpoints:
+                    endpoints.remove(endpoint)
+                EtcdConfiguration.set('/ovs/framework/messagequeue|endpoints', endpoints)
+        except Exception as ex:
+            SetupController._print_log_error('update configurations', ex)
 
         print 'Restarting master node services'
         logger.info('Restarting master node services')
         remaining_nodes = ip_client_map.keys()[:]
         if cluster_ip in remaining_nodes:
             remaining_nodes.remove(cluster_ip)
-        ArakoonInstaller.restart_cluster_remove('ovsdb', remaining_nodes)
+        try:
+            ArakoonInstaller.restart_cluster_remove('ovsdb', remaining_nodes)
+        except Exception as ex:
+            SetupController._print_log_error('restart arakoon cluster', ex)
```

Comment:
Any reason why that would fail? It just restarts all nodes which should always work, right?